mhs/mhs/common/messages/envelope.py

```diff
@@ -0,0 +1,26 @@
+"""This module defines the base envelope used to wrap messages to be sent to a remote MHS."""
+import abc
+import typing
+
+
+class Envelope(abc.ABC):
+    """An envelope that contains a message to be sent to a remote MHS."""
+
+    @abc.abstractmethod
+    def serialize(self):
+        """Produce a serialised representation of this message.
+
+        :return: The serialized representation of this message.
+        """
+        pass
+
+    @classmethod
+    @abc.abstractmethod
+    def from_string(cls, headers: typing.Dict[str, str], message: str):
```

Comment:
Optional: If you want to do:
```suggestion
    def from_string(cls, headers: typing.Dict[str, str], message: str) -> Envelope:
```
then you could do `from __future__ import annotations`. Or you could do:
```suggestion
    def from_string(cls, headers: typing.Dict[str, str], message: str) -> 'Envelope':
```

And same for other classes in this subpackage. Or I could do this as part of RT-72.